### PR TITLE
Make Quartz use PostgreSQL database as the job store

### PR DESCRIPTION
**Pull request by lgonczar**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RPE-141

### Change description ###

Configure Quartz to use the PostgreSQL database as its job data store.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/jobscheduler/config/QuartzConfiguration.java [hunk 1]
```diff
@@ -1,21 +1,35 @@
...
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 
-import javax.inject.Singleton;
 
 @Configuration
+@ConfigurationProperties
 public class QuartzConfiguration {
 
+    private final Map<String, String> quartzProperties = new HashMap<>();
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/jobscheduler/config/QuartzConfiguration.java]: is this automagically set by Spring based on field name?

File: src/main/resources/application.yaml [hunk 2]
```diff
@@ -27,3 +27,22 @@ spring:
...
     validationQuery: SELECT 1
+
+quartzProperties:
+  org.quartz:
+    scheduler:
+      instanceId: AUTO
+    dataSource:
+      jobscheduler:
+        driver: org.postgresql.Driver
+        URL: jdbc:postgresql://${JOB_SCHEDULER_DB_HOST:job-scheduler-database}:${JOB_SCHEDULER_DB_PORT:5432}/jobscheduler
```
Comment (rkondratowicz) [inline src/main/resources/application.yaml]: Maybe there is a way in Spring for one property to reference another? (no idea, genuinely curious)

File: src/main/java/uk/gov/hmcts/reform/jobscheduler/config/QuartzConfiguration.java (hunk 1 above)
Comment (lgonczar) [reply src/main/java/uk/gov/hmcts/reform/jobscheduler/config/QuartzConfiguration.java]: It won't work without the getter, even if the field is public. `@Value` will not work here, either, due to the fact that it's a map.

File: src/main/resources/application.yaml (hunk 2 above)
Comment (lgonczar) [reply src/main/resources/application.yaml]: I'll look into this

Comment (lgonczar) [reply src/main/resources/application.yaml]: Done, it works!

**Review by rkondratowicz — APPROVED** [review]

**Pull request merged**